src/view.tree.js

```diff
@@ -99,19 +113,53 @@ TreeView.prototype.show = function(repo, treeData) {
   }
 }
 
-TreeView.prototype.syncSelection = function() {
-  var tree = this.$view.find('.octotree_view_body').jstree(true)
-    , path = location.pathname
+TreeView.prototype.fetchData = function(opts, success) {
+  // Dummy func
+  throw "It must be implemented from caller when initializing this object"
+}
+
+TreeView.prototype.syncSelection = function(currentPath) {
+  var self = this
+    , tree = this.$view.find('.octotree_view_body').jstree(true)
+    , path = decodeURIComponent(location.pathname)
 
   if (!tree) return
-  tree.deselect_all()
-
-  // e.g. converts /buunguyen/octotree/type/branch/path to path
-  var match = path.match(/(?:[^\/]+\/){4}(.*)/)
-    , nodeId
-  if (match) {
-    nodeId = PREFIX + decodeURIComponent(match[1])
-    tree.select_node(nodeId)
-    tree.open_node(nodeId)
+
```

Comment:
The code below is for loading nodes continuously until getting to selected node in GitHub URL.